src/Controller/SecurityController.php

```diff
@@ -57,16 +57,17 @@ public function editUser(
             throw $this->createAccessDeniedException('vous devez etre connecté');
         }
         $userService = new UserService();
+        $userService->setNewUsername($user->getUsername());
         $form = $this->createForm(EditUserType::class, $userService);
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
             $isUsernameValid = $userService->validateNewUsername($user);
             $isPasswordValid = $userService->validateNewPassword($user, $passwordEncoder);
 
-            $errors = $userService->findUserErrors($user, $passwordEncoder, $isUsernameValid, $isPasswordValid);
+            $errors = $userService->findUserErrors($user, $passwordEncoder, $isPasswordValid);
 
-            if ($isUsernameValid || $isPasswordValid) {
+            if (($isUsernameValid || $isPasswordValid) && !$errors) {
                 $entityManager->flush();
                 return $this->redirectToRoute('home');
             } elseif ($errors) {
```

Comment:
It's a strange idea to bypass the Symfony Form component and it's validation tools in a symfony prooject.
You probably should use constraints in the User entity to validate your form edition.